### PR TITLE
Ensure that flag_arrays are boolean

### LINKED ISSUE #1039: The expected_type of the flag_array on UVData and UVCal objects is `(int, numpy.integer)` instead of `bool`
The expected_type of the flag_array on UVData and UVCal objects is `(int, numpy.integer)` even though they are being set to `bool` because in the `_get_generic_type` method `issubclass(bool, (int, np.integer))` is True.

Is this the expected behavior? I would have thought that we'd want it to be a `bool` so that flag_arrays filled with integers would not pass the check.

I think we might want to add `(bool, np.bool_)` at the top of the for loop over types in `_get_generic_type`. If I do that, the flag_array expected_type becomes `(bool, np.bool_)` and all the existing tests pass.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

- Added booleans to parameter types and fixed equality checking for arrays with strict types.
- Added tests that integer flag arrays fail checks.
- Some test fixture cleanup in test_uvdata.py

## Motivation and Context

fixes #1039 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [x] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

### CODE REVIEW

File: pyuvdata/parameter.py [hunk 1]
```diff
@@ -192,6 +176,34 @@ def __eq__(self, other):
...
+                            f"{self.name} parameter has incompatible dtypes. Left is "
+                            f"{self.value.dtype}, right is {other.expected_type}"
+                        )
+                        return False
+                elif other.strict_type:
+                    # types must match in the other direction
+                    if not isinstance(other.value.item(0), self.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible dtypes. Left is "
+                            f"{other.value.dtype}, right is {other.expected_type}"
```
Comment (mkolopanis) [inline pyuvdata/parameter.py:203]: We're using `other` for both left and right here. Should right be `self.expected_type`?

File: pyuvdata/parameter.py [hunk 2]
```diff
@@ -225,6 +237,24 @@ def __eq__(self, other):
...
+                            f"{self.name} parameter has incompatible types. Left is "
+                            f"{type(self.value)}, right is {other.expected_type}"
+                        )
+                        return False
+                if other.strict_type:
+                    # types must match in the other direction
+                    if not isinstance(other.value, self.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible types. Left is "
+                            f"{type(other.value)}, right is {other.expected_type}"
```
Comment (mkolopanis) [inline pyuvdata/parameter.py:254]: same comment here about `other` being both left and right.

**Review by mkolopanis — COMMENTED** [review]
This looks really good. Thanks for cleaning up the tests as well. I have a question about some docstrings but should be good to go after that.

File: pyuvdata/uvdata/tests/test_uvdata.py [hunk 3]
```diff
@@ -255,6 +255,28 @@ def hera_uvh5_split(hera_uvh5_split_main):
     yield uv1_copy, uv2_copy, uvfull_copy
 
 
+@pytest.fixture(scope="session")
+def hera_uvh5_xx_main():
+    """Read in a HERA uvh5 file."""
+    hera_uvh5_xx = UVData()
+    hera_uvh5_xx.read_uvh5(os.path.join(DATA_PATH, "zen.2457698.40355.xx.HH.uvcA.uvh5"))
+
+    return hera_uvh5_xx
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply pyuvdata/uvdata/tests/test_uvdata.py:264]: me neither, happy to add some cleanup.

File: pyuvdata/parameter.py (hunk 2 above)
Comment (bhazelton) [reply pyuvdata/parameter.py:254]: Yes, I fixed this one and the one before to be more clear.

**Comment by bhazelton** [comment]
@mkolopanis I think I addressed your comments, but now the linter and doctest are failing on the initial linux spinup. I noticed that those two jobs have the os specified to be "linux" whereas the regular tests use a docker image.

**Comment by mkolopanis** [comment]
Interesting that we specifically set the OS there, I would think we would be agnostic to it. Maybe they are older copy/pasta?

File: pyuvdata/parameter.py [hunk 4]
```diff
@@ -225,6 +238,25 @@ def __eq__(self, other):
...
                     return False
             else:
+                # check to see if strict types are used
+                if self.strict_type:
+                    # types must match
+                    if not isinstance(self.value, other.expected_type):
+                        print(
+                            f"{self.name} parameter has incompatible types. Left "
+                            f"requires {type(self.value)}, right is "
+                            "{other.expected_type}"
```
Comment (mkolopanis) [inline pyuvdata/parameter.py:248]: looks like we're missing an `f` here?
```suggestion
                            f"{other.expected_type}"
```

**Comment by bhazelton** [comment]
@mkolopanis I fixed the CI issues on this PR

**Pull request merged**